[
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/799556146",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/21434#issuecomment-799556146",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/21434",
    "id": 799556146,
    "node_id": "MDEyOklzc3VlQ29tbWVudDc5OTU1NjE0Ng==",
    "user": {
      "login": "practicalswift",
      "id": 7826565,
      "node_id": "MDQ6VXNlcjc4MjY1NjU=",
      "avatar_url": "https://avatars.githubusercontent.com/u/7826565?u=5ff13e375b40ea55ecd3c108337dd6a23f68eddf&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/practicalswift",
      "html_url": "https://github.com/practicalswift",
      "followers_url": "https://api.github.com/users/practicalswift/followers",
      "following_url": "https://api.github.com/users/practicalswift/following{/other_user}",
      "gists_url": "https://api.github.com/users/practicalswift/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/practicalswift/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/practicalswift/subscriptions",
      "organizations_url": "https://api.github.com/users/practicalswift/orgs",
      "repos_url": "https://api.github.com/users/practicalswift/repos",
      "events_url": "https://api.github.com/users/practicalswift/events{/privacy}",
      "received_events_url": "https://api.github.com/users/practicalswift/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2021-03-15T16:25:45Z",
    "updated_at": "2021-03-15T16:26:11Z",
    "author_association": "CONTRIBUTOR",
    "body": "Warm welcome as a contributor @sebastianrapp!\r\n\r\nInteresting find! Can you describe how to reproduce the issue locally (what OS, what commands to issue, etc.)?\r\n\r\nMay I ask how you found this issue? It is always interesting to know.",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/799556146/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/799673816",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/21434#issuecomment-799673816",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/21434",
    "id": 799673816,
    "node_id": "MDEyOklzc3VlQ29tbWVudDc5OTY3MzgxNg==",
    "user": {
      "login": "sebastianrapp",
      "id": 160061,
      "node_id": "MDQ6VXNlcjE2MDA2MQ==",
      "avatar_url": "https://avatars.githubusercontent.com/u/160061?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/sebastianrapp",
      "html_url": "https://github.com/sebastianrapp",
      "followers_url": "https://api.github.com/users/sebastianrapp/followers",
      "following_url": "https://api.github.com/users/sebastianrapp/following{/other_user}",
      "gists_url": "https://api.github.com/users/sebastianrapp/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/sebastianrapp/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/sebastianrapp/subscriptions",
      "organizations_url": "https://api.github.com/users/sebastianrapp/orgs",
      "repos_url": "https://api.github.com/users/sebastianrapp/repos",
      "events_url": "https://api.github.com/users/sebastianrapp/events{/privacy}",
      "received_events_url": "https://api.github.com/users/sebastianrapp/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2021-03-15T19:00:12Z",
    "updated_at": "2021-03-15T19:00:12Z",
    "author_association": "NONE",
    "body": "Thank you very much.\r\nI was using macOS Big Sur (version 11.2.1) to reproduce this issue. Here I just need to run bitcoind and got the error message \"Init: Not enough file descriptors available.\"\r\nSo to be honest I was wondering why this issues was not observed before as this is a pretty straightforward use case, right. So maybe I have any special setting in my system of which I am currently not aware.",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/799673816/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/800908131",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/21434#issuecomment-800908131",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/21434",
    "id": 800908131,
    "node_id": "MDEyOklzc3VlQ29tbWVudDgwMDkwODEzMQ==",
    "user": {
      "login": "practicalswift",
      "id": 7826565,
      "node_id": "MDQ6VXNlcjc4MjY1NjU=",
      "avatar_url": "https://avatars.githubusercontent.com/u/7826565?u=5ff13e375b40ea55ecd3c108337dd6a23f68eddf&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/practicalswift",
      "html_url": "https://github.com/practicalswift",
      "followers_url": "https://api.github.com/users/practicalswift/followers",
      "following_url": "https://api.github.com/users/practicalswift/following{/other_user}",
      "gists_url": "https://api.github.com/users/practicalswift/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/practicalswift/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/practicalswift/subscriptions",
      "organizations_url": "https://api.github.com/users/practicalswift/orgs",
      "repos_url": "https://api.github.com/users/practicalswift/repos",
      "events_url": "https://api.github.com/users/practicalswift/events{/privacy}",
      "received_events_url": "https://api.github.com/users/practicalswift/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2021-03-17T08:49:03Z",
    "updated_at": "2021-03-17T08:49:03Z",
    "author_association": "CONTRIBUTOR",
    "body": "I don't have any Mac around to test on unfortunately: anyone running macOS who is able to reproduce this?",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/800908131/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/801029196",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/21434#issuecomment-801029196",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/21434",
    "id": 801029196,
    "node_id": "MDEyOklzc3VlQ29tbWVudDgwMTAyOTE5Ng==",
    "user": {
      "login": "laanwj",
      "id": 126646,
      "node_id": "MDQ6VXNlcjEyNjY0Ng==",
      "avatar_url": "https://avatars.githubusercontent.com/u/126646?u=13b19d1ff2f5f914e180c41418f451a4ba6f8bd1&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/laanwj",
      "html_url": "https://github.com/laanwj",
      "followers_url": "https://api.github.com/users/laanwj/followers",
      "following_url": "https://api.github.com/users/laanwj/following{/other_user}",
      "gists_url": "https://api.github.com/users/laanwj/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/laanwj/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/laanwj/subscriptions",
      "organizations_url": "https://api.github.com/users/laanwj/orgs",
      "repos_url": "https://api.github.com/users/laanwj/repos",
      "events_url": "https://api.github.com/users/laanwj/events{/privacy}",
      "received_events_url": "https://api.github.com/users/laanwj/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2021-03-17T12:05:20Z",
    "updated_at": "2021-03-17T12:09:56Z",
    "author_association": "MEMBER",
    "body": "It is kind of strange, as functions such as `open` `read` `write` return or take an `int` >= 0 there would be no possible reason for there ever to be more than 2**31 file descriptors, so I cannot see the rationale of using a larger type for the limit. That said if this is necessary on some platforms it's necessary.\r\n\r\nEdit: well strictly if you want to use file descriptor 0x7fff_ffff you do need a type that can represent 0x8000_0000 which is > int  \u2026 using that one extra fd doesn't seem worth the potential edge cases introduced by making this function return a different type though :smile: ",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/801029196/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  }
]